src/Facets/CalldataVerificationFacet.sol

```diff
@@ -100,6 +100,35 @@ contract CalldataVerificationFacet {
         );
     }
 
+    // @notice Extracts the non-EVM address from the calldata
+    // @param data The calldata to extract the non-EVM address from
+    // @return nonEVMAddress The non-EVM address extracted from the calldata
+    function extractNonEVMAddress(
+        bytes calldata data
+    ) external pure returns (bytes32 nonEVMAddress) {
+        bytes memory callData = data;
+        ILiFi.BridgeData memory bridgeData = _extractBridgeData(data);
+
+        if (
+            bytes4(data[:4]) == StandardizedCallFacet.standardizedCall.selector
+        ) {
+            // standardizedCall
+            callData = abi.decode(data[4:], (bytes));
```

Comment:
just wondering....would the following code be the same and if no, why?

```solidity
callData = data[4:];
```

Reply:
Yes this would be the same I believe.

Reply:
So when you wrap the calldata in this function call it actually adds two extra 32 bytes words
1. The new offset for the header of the bytes
2. The number of total bytes being passed as a parameter. `abi.decode` knows this so will take the correct value and push it into `callData`